worker-workflow/src/main/java/com/github/cafdataprocessing/WorkflowEvaluator.java

```diff
@@ -0,0 +1,70 @@
+ * limitations under the License.
+ */
+package com.github.cafdataprocessing;
+
+import com.hpe.caf.worker.document.DocumentWorkerConstants;
+import com.hpe.caf.worker.document.JavaScriptDocumentPostProcessor;
+import com.hpe.caf.worker.document.exceptions.PostProcessingFailedException;
+import com.hpe.caf.worker.document.model.Document;
+import com.hpe.caf.worker.document.model.ResponseOptions;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Evaluates a JavaScript workflow against a document, updating document based on matched actions in the workflow
+ */
+public class WorkflowEvaluator
+{
+    /**
+     * Evaluates the provided document against the JavaScript workflow and records the workflow storage reference on
+     * the document for execution in post processing.
+     * @param document The document to evaluate against the workflow. May be modified by evaluation
+     *                 e.g. fields indicating action being executed, the queue to send document to, the post processing
+     *                 script to execute.
+     * @param workflowAsJavaScript JavaScript representation of a workflow.
+     * @param workflowStorageRef storage reference for the workflow passed. Will be recorded as a post processing script
+     *                           if none is already set.
+     * @throws PostProcessingFailedException if there is a failure in workflow script execution.
+     */
+    public static void evaluate(Document document, String workflowAsJavaScript, String workflowStorageRef)
+            throws PostProcessingFailedException
+    {
+        final JavaScriptDocumentPostProcessor jsPostProcessor = new JavaScriptDocumentPostProcessor(workflowAsJavaScript);
+        jsPostProcessor.postProcessDocument(document);
+        setWorkflowPostProcessingScript(document, workflowStorageRef);
+    }
+
+    /**
+     * Sets the post processing on provided document to use workflow in provided storage reference.
+     * If a workflow for post processing is already set on the document then it will be replaced with the new value.
+     * @param document Document to set post processing information on.
+     * @param workflowStorageReference DataStore reference for a workflow that can be executed in post processing.
+     */
+    private static void setWorkflowPostProcessingScript(Document document, String workflowStorageReference){
+        final ResponseOptions responseOptions = document.getTask().getResponseOptions();
+        if(responseOptions.getCustomData()==null){
+            final Map<String, String> customData = new HashMap<>();
+            customData.put(DocumentWorkerConstants.POST_PROCESSING_SCRIPT_CUSTOM_DATA, workflowStorageReference);
+            responseOptions.setCustomData(customData);
+        }
+        else {
+            responseOptions.getCustomData().put(DocumentWorkerConstants.POST_PROCESSING_SCRIPT_CUSTOM_DATA, workflowStorageReference);
```

Comment:
This line is effectively a duplicate of line 61.  If you deal with the customData null potential first, by setting an empty map, then it wouldn't need to be duplicated.

Reply:
If customData is set it looks to be unmodifiable (on ResponseImpl) so I've changed this to copy all the existing customData values (if it is not null) to an empty map, add the post processing script and set the custom data on response options.